### PR TITLE
Mae

### PR DESCRIPTION
**Pull request by pjreddie**
Adding in MAE training module and script and relevant classes to enable it

### CODE REVIEW

File: helios/nn/mae.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+
+class MAE(nn.Module, DistributedMixins):
+    """Masked Auto-Encoder Module."""
+
+    def __init__(
+        self,
+        encoder: nn.Module,
+        decoder: nn.Module,
+        reconstructor: nn.Module,
```
Comment (gabrieltseng) [inline helios/nn/mae.py:26]: nit: This is not exactly analogous to (e.g.) SatMAE since there is now an additional linear layer in the decoder before the reconstructor.
I am not sure this matters but just making a note.

Comment (pjreddie) [reply helios/nn/mae.py:26]: noted, complexity-wise, an extra linear layer with no norm doesn't change theoretical model complexity (it could be merged into the next weighted layer), but it may change training dynamics (probably not too much)

File: scripts/base_debug_scripts/galileo.py
Comment (pjreddie) [inline scripts/base_debug_scripts/galileo.py:1]: this file shouldn't actually be changed, I'll roll this back, not sure where this came from

File: helios/train/loss.py [hunk 2]
```diff
@@ -320,6 +322,62 @@ def compute(
...
+        for modality in predictions.modalities:
+            if getattr(predictions, modality) is not None:
+                masked_name = targets.get_masked_modality_name(modality)
+                valid_dict[modality] = getattr(targets, modality)
+                valid_dict[masked_name] = getattr(targets, masked_name)
+        valid_targets = TokensAndMasks(**valid_dict)
+        labels, label_masks = self._flatten_helios_data(valid_targets)
+        data = data * (masks == MaskValue.DECODER.value)
+        labels = labels * (masks == MaskValue.DECODER.value)
+        return F.mse_loss(data, labels)
```
Comment (gabrieltseng) [inline helios/train/loss.py]: Fix averaging

File: helios/train/loss.py [hunk 3]
```diff
@@ -320,6 +322,62 @@ def compute(
...
+        """
+        data, masks = self._flatten_helios_data(predictions)
+        valid_dict = {}
+        for modality in predictions.modalities:
+            if getattr(predictions, modality) is not None:
+                masked_name = targets.get_masked_modality_name(modality)
+                valid_dict[modality] = getattr(targets, modality)
+                valid_dict[masked_name] = getattr(targets, masked_name)
+        valid_targets = TokensAndMasks(**valid_dict)
+        labels, label_masks = self._flatten_helios_data(valid_targets)
```
Comment (gabrieltseng) [inline helios/train/loss.py:375]: Use label masks (& / or make sure encoder masks are unchanged)

File: helios/train/callbacks/speed_monitor.py [hunk 4]
```diff
@@ -26,6 +27,14 @@ def pre_train(self) -> None:
         train_module = self.trainer.train_module
 
         self._token_budget = self.trainer.data_loader.token_budget
+        if isinstance(train_module, MAETrainModule):
```
Comment (gabrieltseng) [inline helios/train/callbacks/speed_monitor.py]: This can be in line 38 isinstance(train_module, (MAETrainModule, Lat...))

File: helios/train/loss.py (hunk 3 above)
Comment (gabrieltseng) [reply helios/train/loss.py:375]: Add to encoder integration test: https://github.com/allenai/helios/issues/161

Comment (pjreddie) [reply helios/train/loss.py:375]: added this test for Mae module, it's harder for encoder since you'd have to reconstruct the original mask (as we do in MAE). but the encoder in MAE doesn't modify the mask according to the tests
